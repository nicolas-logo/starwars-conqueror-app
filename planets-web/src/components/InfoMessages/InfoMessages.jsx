// eslint-disable-next-line react/prop-types
export const InfoMessages = ({ apiErrorMessage, loading }) => (
    <div data-testid="info-messages">
        {apiErrorMessage !== null && <span className='text-danger'>{apiErrorMessage}</span>}
        {loading && !apiErrorMessage && <h4 className='text-white'>Loading...</h4>}
    </div>
)
