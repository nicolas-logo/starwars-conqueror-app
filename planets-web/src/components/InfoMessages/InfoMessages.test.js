import '@testing-library/jest-dom/extend-expect'
import { render, cleanup } from '@testing-library/react'
import { InfoMessages } from './InfoMessages'
import renderer from 'react-test-renderer'

afterEach(() => {
  cleanup()
})

describe('InfoMessages tests', () => {
  it('should renders error message when `apiErrorMessage` prop is not null', () => {
    const apiErrorMessage = 'Error: Unable to retrieve photos.'
    const { getByText } = render(<InfoMessages apiErrorMessage={apiErrorMessage} />)
    const errorMessageElement = getByText(apiErrorMessage)
    expect(errorMessageElement).toBeInTheDocument()
    expect(errorMessageElement).toHaveClass('text-danger')
  })

  it('should renders "Loading..." message when `loading` prop is true', () => {
    const loading = true
    const { getByText } = render(<InfoMessages loading={loading} />)
    const loadingMessage = getByText('Loading...')
    expect(loadingMessage).toBeInTheDocument()
    expect(loadingMessage).toHaveClass('text-white')
  })

  it('should should match the snapshot', () => {
    const tree = renderer.create((<InfoMessages loading={false} imagesLength={0} />)).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
