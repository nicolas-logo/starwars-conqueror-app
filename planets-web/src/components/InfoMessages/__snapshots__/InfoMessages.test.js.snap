// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoMessages tests should should match the snapshot 1`] = `
<div
  data-testid="info-messages"
>
  <span
    className="text-danger"
  />
</div>
`;
